src/Symfony/Component/Asset/Preload/PreloadManager.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\Preload;
+
+/**
+ * Manages preload HTTP headers.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PreloadManager implements PreloadManagerInterface
```

Comment:
I meant decoration by inheritance of course.
My root issue is that final here would be only forbidding use cases while providing no benefit for us.
"final" should only used on method/classes that are not bound by any contract - like data objects.
When an interface covers the said methods, the contract is already enforced and inheritance should not be prevented.